### PR TITLE
Add _mailer suffix to generated file names.

### PR DESCRIPTION
**Pull request by caike**
It is somewhat confusing that [controllers](https://github.com/rails/rails/blob/master/railties/lib/rails/generators/rails/controller/controller_generator.rb#L10) and [jobs](https://github.com/rails/rails/blob/master/activejob/lib/rails/generators/job/job_generator.rb#L19) follow this convention, but [mailers](https://github.com/rails/rails/blob/master/actionmailer/lib/rails/generators/mailer/mailer_generator.rb#L10) don't.

### CODE REVIEW

**Comment by rafaelfranca** [comment]
The implementation looks good to me. @dhh what do you think about the convention change?

**Comment by dhh** [comment]
:+1:

**Comment by rafaelfranca** [comment]
Great! @caike mind to add a CHANGELOG entry and squash your commits?

File: actionmailer/lib/rails/generators/mailer/mailer_generator.rb [hunk 1]
```diff
@@ -4,16 +4,22 @@ class MailerGenerator < NamedBase
...
         if self.behavior == :invoke
           template "application_mailer.rb", 'app/mailers/application_mailer.rb'
         end
       end
 
       hook_for :template_engine, :test_framework
+
+      protected
+        def file_name
+          @_file_name ||= super.gsub(/\_mailer/i, '')
```
Comment (JuanitoFatas) [inline actionmailer/lib/rails/generators/mailer/mailer_generator.rb]: Can we use `sub` here?

```ruby
@_file_name ||= super.sub(/\_mailer\.rb$/i, '')
```

Comment (caike) [inline actionmailer/lib/rails/generators/mailer/mailer_generator.rb]: I've tried your suggestion @JuanitoFatas , but looks like using `sub` this way doesn't account for removing duplicates. See the failure on _MailerGeneratorTest#test_mailer_suffix_is_not_duplicated_ at https://travis-ci.org/rails/rails/jobs/46081505

**Comment by carlosantoniodasilva** [comment]
:green_heart::yellow_heart::heart::blue_heart::purple_heart:

File: railties/test/generators/mailer_generator_test.rb [hunk 2]
```diff
@@ -25,55 +25,55 @@ def test_application_mailer_skeleton_is_created
 
   def test_mailer_with_i18n_helper
     run_generator
-    assert_file "app/mailers/notifier.rb" do |mailer|
+    assert_file "app/mailers/notifier_mailer.rb" do |mailer|
       assert_match(/en\.notifier\.foo\.subject/, mailer)
       assert_match(/en\.notifier\.bar\.subject/, mailer)
```
Comment (carlosantoniodasilva) [inline railties/test/generators/mailer_generator_test.rb]: Shouldn't the i18n path for the subject change as well?

Comment (caike) [inline railties/test/generators/mailer_generator_test.rb]: Good catch! :+1: 
New PR submitted

**Comment by caike** [comment]
Thanks for catching those @y-yagi ! :+1:

**Pull request merged**